cli/src/utils.rs

```diff
@@ -80,24 +80,28 @@ impl PlatformSelector {
                 // defaults based on previous selections if previous selections are not empty
                 // otherwise, selects all
                 let all_ios_archs = IosArch::all_strings();
-                let defaults: Vec<bool> = if config.ios.is_empty() {
+                let defaults: Vec<bool> = if config.ios.is_none() {
                     vec![true; all_ios_archs.len()]
                 } else {
                     all_ios_archs
                         .iter()
-                        .map(|&acrch| config.ios.contains(acrch))
+                        .map(|&acrch| config.ios.as_ref().unwrap().contains(acrch))
                         .collect()
                 };
 
                 // clear previous selections before update
-                config.ios.clear();
+                if let Some(ref mut ios) = config.ios {
```

Comment:
We can just new a `ios` here, assigning from `config.ios` and then clear it is not necessary.

Reply:
the same as `android` below

Reply:
[029aaec](https://github.com/zkmopro/mopro/pull/417/commits/029aaec3858a5897fbbc747d4efea1752807973e)